android/engine/src/main/java/org/smartregister/fhircore/engine/task/FhirCarePlanGenerator.kt

```diff
@@ -99,6 +104,11 @@ constructor(
     data: Bundle = Bundle(),
     generateCarePlanWithWorkflowApi: Boolean = false,
   ): CarePlan? {
+    val relatedEntityLocationTags =
+      subject.meta.tag.filter {
+        it.system == context.getString(R.string.sync_strategy_related_entity_location_system)
```

Comment:
I would not expect `system` to be unique, wouldn't it be code that is the unique item to filter on? Isn't [meta.tag a coding](https://hl7.org/fhir/R4B/resource.html#Meta), or is this the wrong reference?

Reply:
The idea here is we get all the tags using the system for `RelatedEntityLocation` coding. In the current use case, this should return a single tag.

Reply:
At this point, propagating the locationId that's used as the code is not so much useful

Reply:
The issue is that `system` means the terminology system, like ICD11 or something, even though we're not using it that way, so this wouldn't return the correct set of tags if we used it correctly.

So I guess I'm suggesting 2 changes (that contradict what's written in the issue, apologies that I did not flag this earlier)

1. We use system correctly, so change this to "https://smartregister.org/" if we say we are using our own internal terminology system
2. define and store in docs a "code" that means "Related Entity Location" and filter by "code" here

What do you think?

cc @dubdabasoduba

Reply:
I see. This means we'd have to update the other tags as well with this interpretation of the system.

Reply:
Sample of all meta tags.

```json
"tag": [
      {
        "system": "urn:oid:2.16.578.1.12.4.1.1.8655",
        "display": "Patient Registration",
        "_display": {
          "extension": [
            {
              "url": "http://hl7.org/fhir/StructureDefinition/translation",
              "extension": [
                {
                  "url": "lang",
                  "valueCode": "id"
                },
                {
                  "url": "content",
                  "valueString": "Pendaftaran Pasien"
                }
              ]
            }
          ]
        }
      },
      {
        "system": "https://smartregister.org/related-entity-location-tag-id",
        "code": "33f45e09-f96e-41d3-9916-fb96455a4cb2",
        "display": "Related Entity Location"
      },
      {
        "system": "https://smartregister.org/care-team-tag-id",
        "code": "47d68cac-306f-4b75-9704-b4ed48b24f76",
        "display": "Practitioner CareTeam"
      },
      {
        "system": "https://smartregister.org/location-tag-id",
        "code": "Not defined",
        "display": "Practitioner Location"
      },
      {
        "system": "https://smartregister.org/organisation-tag-id",
        "code": "ca7d3362-8048-4fa0-8fdd-6da33423cc6b",
        "display": "Practitioner Organization"
      },
      {
        "system": "https://smartregister.org/practitioner-tag-id",
        "code": "9db48504-9f63-411b-b61e-28351d7af5e8",
        "display": "Practitioner"
      },
      {
        "system": "https://smartregister.org/app-version",
        "code": "1.1.0-sidBunda",
        "display": "Application Version"
      }
    ]
```

Reply:
Yea I'm just knowing that, I don't want to do that all at once do, we can make sure we do it correctly here and then change the remainder over time. I'm checking in w/ @dubdabasoduba and @f-odhiambo to confirm we're all on the same page...

Reply:
For uniqueness, we've been using a combination of system + code

Reply:
@dubdabasoduba raised good points on why we do it this way and that there's a perspective through which it is semantic, I will write up some docs based on the conversation and resolve this conversation, we can move ahead as is, thank you for the details